### DISCUSSION TITLE
Allow images in popups (sixel protocol?)

### DISCUSSION THREAD

**Discussion by eugenesvk** [helix-editor/helix#6938] [Ideas]
I'd like to have a keybind that shows an image in a popup window, for example, a keymap cheat sheet on <kbd>F1</kbd>
As far as I'm aware, many terminals support a special standard imaging protocol https://www.arewesixelyet.com/ for that allows you to display images in the terminal

Would be great if I could control that behavior via a Helix command

**Comment by pascalkuthe** [comment]
I dont think we want to support feature like that from inside helix. There are multiple different nom-stanrd protocols. None of which are stadarized and usually only support by an emulator or two. Thisncoukd be done by plugins in the future but I dont see any reason why this needs to be integrated into core.

In the meantime.you can simply open images in a seperate window, with a terminal multiplexer or the terminal emulator (all of these usually support both overlaysnand splits)

**Comment by eugenesvk** [comment]
> None of which are stadarized and usually only support by an emulator or two. 

Sixel is supported by a dozen terminals according to https://www.arewesixelyet.com/

**Comment by brunoherrick** [comment]
+1 Adding image preview/ rendering would be particularly useful for markdown notes. I have lots of notes containing images and this would be perfect to avoid using GUI editors/ note apps.

**Comment by jbirnick** [comment]
+1 both for notetaking systems and for math formula preview in typesetting

**Comment by TornaxO7** [comment]
Hm... I could rather see this as a plugin at most.
For me, helix is a terminal editor which should work fully within a terminal
but a terminal doesn't require displaying images like in the tty so I don't see this feature to be in core.

**Comment by sabisho** [comment]
Support for kitty image protocol maybe?
It's better than sixel.
You'll can look at yazi for implementation idea.

**Comment by koalazub** [comment]
I wanted to build out a notebook plugin for helix and did this as an experiment: 
https://github.com/koalazub/helix/pull/1

I was thinking of a more abstracted method of interfacing with protocols. But the point about tty is crucial. I didn't think about that when working on this.

I want it built in so I can continue working on the plugin and preferably getting it functional before my final year of postgrad kicks off and I can not need to rely on Pluto.jl or Marimo and instead lean on something like this. I wrote up my thoughts here: https://github.com/helix-editor/helix/pull/8675

It would be amazing to render math/scientific computation via Helix
